### PR TITLE
Update validation of optional string properties

### PR DESCRIPTION
**Pull request by djino97**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/779

### CODE REVIEW

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs [hunk 1]
```diff
@@ -47,16 +52,21 @@ public class PatchDbUserMapper : IPatchDbUserMapper
...
+                    continue;
+                }
+                if (item.path.EndsWith(nameof(EditUserRequest.City), StringComparison.OrdinalIgnoreCase))
+                {
+                    result.Operations.Add(new Operation<DbUser>(item.op, item.path, item.from, value(item)));
                     continue;
                 }
 
-                result.Operations.Add(new Operation<DbUser>(item.op, item.path, item.from, item.value));
+                    result.Operations.Add(new Operation<DbUser>(item.op, item.path, item.from, item.value));
```
Comment (MariyaSin) [inline src/UserService.Mappers/Models/PatchDbUserMapper.cs:69]: remove tab spase

File: src/UserService.Mappers/Models/UserInfoMapper.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public class UserInfoMapper : IUserInfoMapper
                 Gender = (UserGender)dbUser.Gender,
                 DateOfBirth = dbUser.DateOfBirth?.ToShortDateString(),
                 City = dbUser.City,
-                StartWorkingAt = dbUser.StartWorkingAt?.ToShortDateString(),
+                StartWorkingAt = dbUser.DateOfBirth?.ToShortDateString(),
```
Comment (MariyaSin) [inline src/UserService.Mappers/Models/UserInfoMapper.cs:32]: StartWorkingAt = dbUser.StartWorkingAt?.ToShortDateString()

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs (hunk 1 above)
Comment (djino97) [reply src/UserService.Mappers/Models/PatchDbUserMapper.cs]: fixed

File: src/UserService.Mappers/Models/UserInfoMapper.cs (hunk 2 above)
Comment (djino97) [reply src/UserService.Mappers/Models/UserInfoMapper.cs]: fixed

**Review by MariyaSin — APPROVED** [review]

**Review by BelehovEgor — APPROVED** [review]

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 3]
```diff
@@ -308,7 +308,8 @@ public OperationResultResponse<Guid> Execute(CreateUserRequest request)
 
             var dbUser = _mapperUser.Map(request, avatarImageId);
 
-            var password = request.Password ?? _generatePassword.Execute();
+            string password = !string.IsNullOrEmpty(request.Password) && request.Password.Trim().Any()
```
Comment (TIove) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:311]: Probably you can write it shorter like !string.IsNullOrEmpty(request.Password?.Trim()) and you have not to write &&

Comment (TIove) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs:311]: And same in DbUserMapper.cs on 49 line

**Review by TIove — APPROVED** [review]

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 3 above)
Comment (TIove) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:311]: mb !string.IsNullOrEmpty(request.Password?.Trim())

**Review by borovHokage — APPROVED** [review]

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 3 above)
Comment (djino97) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs]: Reworked

**Pull request merged**